extensions-contrib/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureDataSegmentConfig.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.storage.azure;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.annotation.Nonnull;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Stores the configuration for segments written to Azure deep storage
+ */
+public class AzureDataSegmentConfig
+{
+  @JsonProperty
+  @NotNull
+  private String container;
+
+  @JsonProperty
+  @Nonnull
```

Comment:
Why `Nonnull` when the previous one is annotated with 'NotNull'?

Also is prefix a required config? Why is assigned to an empty string?

Perhaps using a `@JsonCreator` constructor with Precondition checks will make it clearer what is required in each field

```
 @JsonCreator
  public AzureDataSegmentConfig(
      ...
      @JsonProperty("prefix") String prefix)
  {
    this.prefix = Preconditions.checkState(!StringUtils.isEmpty(prefix), "prefix must be non empty");
    ...
  }
```

Then you don't need all the setters

Reply:
prefix is not required. Before adding this option segments were written to the root directory within the segment container specified, in a directory named after the datasource. Do we want to change the behavior here and specify a non empty default prefix? I'm not sure how this change would affect users already using the azure extension whose data is already written, will we not be able to find the segment data in this case?

I will fix to @NotNull

Reply:
fixed